dao/comment.dao.server.js

```diff
@@ -1,28 +1,25 @@
-const commentModel = require('../models/comments.model.server')
-const ObjectID = require("bson-objectid");
+const commentModel = require('../models/comments.model.server');
 
-const findAllComments = () => commentModel.find();
+findAllComments = () => commentModel.find();
 
-const findCommentById = (commentId) => commentModel.findOne({comment_id:commentId});
+findCommentByCommentID = (commentId) => commentModel.find({comment_id:commentId});
 
-const deleteComment = (commentId) => commentModel.deleteOne({comment_id:commentId},(err, obj) => {
-    if (err) throw err;
-    console.log("1 document deleted");
-});
+findCommentByUserID = (userId) => commentModel.find({user_id: userId});
```

Comment:
find will always return a cursor even if no comment found, and thus the condition in delete comment will always be true.

Reply:
There is no different between my code and your code

Reply:
you are using find() instead of findOne()